generator/views.py

```diff
@@ -2,57 +2,118 @@
-        raise NotImplementedError()
+    def from_db_views(
+        klass, app: str, channels: List[Dict[str, str]], db_views: dict
+    ) -> Iterator[View]:
+        """Get Looker views from app."""
+        raise NotImplementedError("Only implemented in subclass.")
+
+    def get_type(self) -> str:
+        """Get the type of this view."""
+        return self.view_type
+
+    def as_dict(self) -> dict:
+        """Get this view as a dictionary."""
+        return {
+            "type": self.view_type,
+            "tables": self.tables,
+        }
+
+    def __str__(self):
+        """Stringify."""
+        return f"name: {self.name}, type: {self.type}, table: {self.tables}"
+
+    def __eq__(self, other) -> bool:
+        """Check for equality with other View."""
+
+        def comparable_dict(d):
+            return {tuple(sorted(t.items())) for t in self.tables}
+
+        if isinstance(other, View):
+            return (
+                self.name == other.name
+                and self.view_type == other.view_type
+                and comparable_dict(self.tables) == comparable_dict(other.tables)
+            )
+        return False
 
 
 class PingView(View):
     """A view on a ping table."""
 
+    type: str = "ping_view"
+
+    def __init__(self, name: str, tables: List[Dict[str, str]]):
+        """Create instance of a PingView."""
+        super().__init__(name, PingView.type, tables)
+
     @classmethod
-    def from_db_views(klass, variants: dict, db_views: dict):
-        """Get Looker views from db views and app variants."""
+    def from_db_views(
+        klass, app: str, channels: List[Dict[str, str]], db_views: dict
+    ) -> Iterator[PingView]:
+        """Get Looker views for a namespace."""
         views = defaultdict(list)
-        for app in variants:
-            if app.get("deprecated"):
-                continue
-            is_release = app.get("app_channel") == "release"
-            dataset_id = app["bq_dataset_family"]
-            for view_id, references in db_views[dataset_id].items():
+        for channel in channels:
+            dataset = channel["dataset"]
+
+            for view_id, references in db_views[dataset].items():
                 if view_id in OMIT_VIEWS:
                     continue
-                table: Dict[str, Union[str, bool]] = {
-                    "table": f"mozdata.{dataset_id}.{view_id}"
-                }
-                if "app_channel" in app:
-                    table["channel"] = app["app_channel"]
-                if len(references) == 1 and references[0][-2] == f"{dataset_id}_stable":
-                    # view references a single table in the stable dataset
-                    table["is_ping_table"] = True
-                elif not is_release:
-                    continue  # ignore non-ping tables from non-release datasets
+
+                table: Dict[str, str] = {"table": f"mozdata.{dataset}.{view_id}"}
+
+                if channel.get("channel"):
```

Comment:
this should use the more idiomatic `in` syntax, with the added benefit that it correctly handles falsey values
```suggestion
                if "channel" in channel:
```